src/meta/src/rpc/server.rs

```diff
@@ -143,6 +142,17 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
             let (leader_info, is_leader) = note_status_leader_rx.borrow().clone();
             let leader_addr = leader_info_to_host_addr(leader_info);
 
+            // Implementation of naive fencing mechanism:
+            // leader nodes should panic if they loose their leader position
+            // Current implementation panics, if leader looses lease.
+            // Alternative implementation that panics only if leader is unable to re-acquire lease:
+            // if was_leader && !is_leader {
+            if was_leader {
```

Comment:
Alright. I will go for that. As a note: With that we now have a slightly different behaviour, since we can do failover from leader to follower if it is caused by election re-runs and not during the term.

Reply:
Doing failover from leader to follower is too dangerous, because the time of graceful shutdown the leader services and sub-tasks is uncertain and likely to exceed the time of the lease expire time.

Reply:
I will introduce another green thread that calls `process::exit` if it notices that node lost leadership. Kind of like before, but seperate from node status green thread.

Reply:
I pushed it in 671a5dd. The only downside now is that we have two places in the code that handle fencing.

Reply:
If the leadership is taken, we should exit the leader logic as soon as possible, it's better to exit as soon as the leader is taken